src/monodroid/jni/monodroid-networkinfo.c

```diff
@@ -107,6 +109,11 @@ _monodroid_get_network_interface_state (const char *ifname, mono_bool *is_up, mo
   leave:
 	if (!ret)
 		log_warn (LOG_NET, "Unable to determine interface '%s' state using Java API", ifname);
+
+	if (networkInterface != NULL && env != NULL) {
```

Comment:
Instead of admitting that `env` could ever be NULL, I'd rather update `ensure_jnienv()` to *assert* that the returned value is not NULL.  `get_jnienv()` must not ever return NULL.

Reply:
it's because of the `gotos`